pkg/api/v1/types.go

```diff
@@ -1429,6 +1429,20 @@ type PodSpec struct {
 	// List of volumes that can be mounted by containers belonging to the pod.
 	// More info: http://releases.k8s.io/HEAD/docs/user-guide/volumes.md
 	Volumes []Volume `json:"volumes,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+	// List of initialization containers belonging to the pod.
+	// Init containers are executed in order prior to containers being started. If any
+	// init container fails, the pod is considered to have failed and is handled according
+	// to its restartPolicy. The name for an init container or normal container must be
+	// unique among all containers.
+	// Init containers may not have Lifecycle actions, Readiness probes, or Liveness probes.
```

Comment:
The more specific type (when so many fields are the same) may end up
being worse for both internal clarity and clients.  We can validate
unsupported fields easily, but it's much harder to deal with two very
similar types with the same code / logical structure.